settings.gradle

```diff
@@ -31,10 +31,6 @@ rootProject.name="spring-boot-build"
 settings.gradle.projectsLoaded {
 	gradleEnterprise {
 		buildScan {
-			if (settings.gradle.rootProject.hasProperty('buildJavaHome')) {
```

Comment:
The idea/hope is that the build scans show the actual toolchain, but we'll see. Couldn't test this really.

Reply:
Here's a scan against `2.4.x` with `-PtoolchainVersion=16`: https://ge.spring.io/s/yadj75gujofi6

Reply:
I wonder why 2.4.x is targeted here? I guess we would need Groovy 3 as well in the 2.4.x mainline to make this work.

Reply:
Support for Java 16 starts with Spring Framework 5.3 which is what Spring Boot 2.4+ uses. We need a GA release of Spring Boot that folks using Java 15 could rely on once 16 is out (given that 15 will be effectively EOL at that time).

As for Groovy 3, perhaps we could upgrade to it on the Java 16 build?

Reply:
I think I would need a little help on getting this to work on 2.4.x - I currently don't see a good way to do this sort of conditional dependency upgrade

Reply:
Alternatively, we could exclude the CLI stuff from JDK 16 support. Much like the Gradle Plugin. After all, the Framework support says "best effort" as well

Reply:
Yeah, @wilkinsona and I reached the same conclusion. The CLI build should be excluded in `2.4.x` when building against 16 indeed. Do you have time to update your PR in that direction?

Reply:
Well, this would only happen for the actual pipelines e.g. via `-x :spring-boot-project:spring-boot-cli:test`. I wouldn't know what I should update in this PR - cluttering every CLI test with `@DisabledForJreRange` seems like an overkill.

Reply:
I'd rather find a way to disable tests for the CLI module altogether with Java 16. When we had a Maven build we used profiles to disable Surefire. I guess Gradle should have a similar feature.

Reply:
If you know one that is also aware of the toolchain version, that would be great. We could also use that for the gradle plugin then. I just don't know how we could exclude a whole module based on the toolchain version - apart from coming up with our own way. Let me know if I should experiment around that.

Reply:
I think we'll have to come up with our own way. Alternatively, we could use `@DisabledForJreRange` all over the place. One advantage of that is the tests are disabled no matter how they're launched.

Reply:
I'm currently writing a custom plugin that lets you specify the maximum compatible java version inside the build.gradle for that particular submodule. The plan is that this information is consumed in order to dynamically disable the things that run with the optional toolchain.